packages/sysutils/busybox/scripts/pastebinit

```diff
@@ -4,8 +4,6 @@
 # Copyright (C) 2009-2016 Stephan Raue (stephan@openelec.tv)
 # Copyright (C) 2017-present Team LibreELEC (https://libreelec.tv)
 
-if [ -n "${PASTEUSR}" -a "${PASTEPWD}" ]; then
-  cat "$@" | curl -F 'f:1=<-' ${PASTEUSR}:${PASTEPWD}@ix.io
-else
-  cat "$@" | curl -F 'f:1=<-' http://ix.io
-fi
+token=$(echo "@DISTRONAME@-@OS_VERSION@" | sha256sum | cut -d ' ' -f 1)
```

Comment:
Why not only have the token replacement here and do the processing in the build step?

Reply:
while the token is not secret or secure in any way (obviously) this avoids having it somewhere in cleartext
I know that real world implication are rather low but calculation at every send is cheap and additional it removes any deps at the compile host.

better ideas welcome :)

Reply:
Obscurity isn't security.

Is there any plan to drop tokens? What if a token needs to be dropped, does that mean a specific release cannot use the Pastebin service anymore?

Reply:
This is not meant to be a fully secure feature, just an additional hurdle to stop random Joe Average from easily abusing our server as a public dump-site for (potentially problematic) stuff which we then have to care about (disk space, DMCA/GDPR removal requests etc).

In case of excessive abuse we could remove tokens, in that case obviously the affected releases won't be able to upload logs anymore - but that wouldn't be too different to the current situation with eg ix.io stopping operations.

This might help us to provide some kind of service (eg switching to short-term tokens for the latest nightly builds) if the maintenance effort starts getting too high - still better than having to completely shut down the service.

Reply:
Kodi doesn't have these issues with their service. I'd imagine they get a lot more pastes than we will.

Reply:
Better be safe than sorry - preparing for the (not really) unexpected is always a good idea

Reply:
Why not pull it out into an add-on that can be updated outside of the base image?

Reply:
We need something in the image so users can ssh in and provide logs if kodi didn't start.

I really don't want to overcomplicate this, it can always be extended later if/when problems arise